pkg/venafi/tpp/connector.go

```diff
@@ -1256,6 +1279,27 @@ func (c *Connector) RetrieveCertificate(req *certificate.Request) (certificates
 	}
 }
 
+// After many tests, we found that the only way to know if the current
+// certificate request is stuck due to an old failed enrollment is to look for
+// "WebSDK CertRequest" or "Fix any errors, and then click Retry." in the
+// retrieve response when it returns a 500.
+func shouldReset(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	if !strings.Contains(err.Error(), "Status: 500") {
+		return false
+	}
+
+	if strings.Contains(err.Error(), "This certificate cannot be processed while it is in an error state. Fix any errors, and then click Retry.") ||
```

Comment:
This is good. It is bit thorough checking. 
Just suggestion may be we can only search strings.Contains("error state") in first condition.  It might not be require if TPP will not be changing error.  No blocker.

Reply:
I agree, we could use a shorter substring to prevent potential future breakages if TPP changes this error message. I can change that in a later PR (this PR has being going for a while now).